lib/logstash/outputs/firehose.rb

```diff
@@ -43,12 +43,19 @@ class LogStash::Outputs::Firehose < LogStash::Outputs::Base
   TEMPFILE_EXTENSION = "txt"
   FIREHOSE_STREAM_VALID_CHARACTERS = /[\w\-]/
 
+  # These are hard limits
+  FIREHOSE_PUT_BATCH_SIZE_LIMIT = 4_000_000 # 4MB
```

Comment:
would be nice to have these properties configurable

Reply:
This is a hard limit from AWS, you cannot change it so I'm not sure the value in making it configurable.